src/App.tsx

```diff
@@ -1,61 +1,41 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { Todo } from './types/Todo';
+import { AddTodoForm } from './components/AddTodoForm/AddTodoForm';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const getUsersById = (id: number) =>
+  usersFromServer.find(user => user.id === id);
+
+const initialTodos: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: getUsersById(todo.userId),
+  };
+});
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const getLargestId = () => {
+    const id = todos.map(todo => todo.id);
+
+    return Math.max(...id);
+  };
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(current => [...current, newTodo]);
+  };
 
-export const App = () => {
   return (
     <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <h1 className="title main_title">Add todo form</h1>
+
+      <AddTodoForm addTodo={addTodo} getLargestId={getLargestId} />
```

Comment:
Why are u pass it like props?
Move this helper function to a separate file, and pass arguments to it
`const getLargestId = (todos: Todo[]) => {`